hw/mcu/nxp/kinetis/src/hal_timer.c

```diff
@@ -0,0 +1,428 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <stdlib.h>
+#include <assert.h>
+
+#include "os/mynewt.h"
+#include "hal/hal_timer.h"
+#include "mcu/cmsis_nvic.h"
+
+#include "fsl_lptmr.h"
+
+#define KINETIS_TIMERS_MAX 2
+
+#if (FSL_FEATURE_LPTMR_CNR_WIDTH_IS_32B != 0)
+#error "The Kinetis hal_timer driver currently only supports 16-bit timers"
+#endif
+
+#if MYNEWT_VAL(TIMER_1) && (KINETIS_TIMERS_MAX < 2)
+#error "This MCU does not support TIMER_1"
+#endif
+
+/*
+ * This value for MCGIRCLK matches the current configuration for Kinetis BSPs
+ * where the slow clock is used and the divider is set to 1 (FCRDIV==0).
+ */
+#define MCGIRCLK_HZ 32768
+
+struct kinetis_hal_tmr {
+    LPTMR_Type *base;
+    uint16_t overflow; /* MSB of the current counter is handled manually */
+    TAILQ_HEAD(hal_timer_qhead, hal_timer) hal_timer_q;
+};
+
+#if MYNEWT_VAL(TIMER_0)
+struct kinetis_hal_tmr kinetis_tmr0 = {
+    .base = LPTMR0,
+};
+#endif
+#if MYNEWT_VAL(TIMER_1)
+struct kinetis_hal_tmr kinetis_tmr1 = {
+    .base = LPTMR1,
+};
+#endif
+
+static IRQn_Type kinetis_timer_irqs[] = LPTMR_IRQS;
```

Comment:
**kinetis_timer_irqs** could be **const**, maybe **kinetis_timers** also

Reply:
Done.